PEP 8104: 2023 Term Steering Council election (#2875)
python/peps: pep-8104.rst

PEP: 8104
Title: 2023 Term Steering Council election
Author: Ee Durbin <ee@python.org>
Sponsor: Brett Cannon <brett@python.org>
Status: Active
Type: Informational
Content-Type: text/x-rst
Created: 08-Nov-2022

Abstract
========

This document describes the schedule and other details of the December
2022 election for the Python steering council, as specified in
:pep:`13`. This is the steering council election for the 2023 term
(i.e. Python 3.12).

Election Administration
=======================

The steering council appointed the
`Python Software Foundation <https://www.python.org/psf-landing/>`__
Director of Infrastructure, Ee Durbin, to administer the election.

Schedule
========

There will be a two-week nomination period, followed by a two-week
vote.

The nomination period shall be: November 14, 2022 through `November 28, 2022 AoE
<https://www.timeanddate.com/worldclock/fixedtime.html?msg=Python+Steering+Council+nominations+close&iso=20221129T00>`_ [#note-aoe]_.

> The voting period shall be: December 1, 2022 through `December 14, 2022 AoE

Reviewer: Its not *that* important since its the start date, not the end date. However, as mentioned previously this is potentially confusing, since AoE is only [defined](https://www.ieee802.org/16/aoe.html) and used for _deadlines_, not start times, and arguably the most logical extension of the concept to start times would imply a voting start time of 2022-11-30 10:00 UTC, rather than the 2022-12-01 12:00 UTC (26 hours later) that is listed below.

Reviewer 2: It doesn't matter as folks can't vote until they get the email anyway, and once they do they can start. The start time is inconsequential as there's a physical blocker dictating when people can begin.

PEP: 8104
Title: 2023 Term Steering Council election
Author: Ee Durbin <ee@python.org>
Sponsor: Brett Cannon <brett@python.org>
Status: Active
Type: Informational
Content-Type: text/x-rst
Created: 08-Nov-2022

Abstract
========

This document describes the schedule and other details of the December
2022 election for the Python steering council, as specified in
:pep:`13`. This is the steering council election for the 2023 term
(i.e. Python 3.12).

Election Administration
=======================

The steering council appointed the
`Python Software Foundation <https://www.python.org/psf-landing/>`__
Director of Infrastructure, Ee Durbin, to administer the election.

Schedule
========

There will be a two-week nomination period, followed by a two-week
vote.

> The nomination period shall be: November 14, 2022 through `November 29, 2022 12:00 UTC

Reviewer 3: Should the nomination start have a time?

And if November 14, 12:00 UTC to November 29, 12:00 UTC, that's 15 days.

> There will be a two-week nomination period...

Should it be 14 days?

Reviewer 2: It's Nov 14 - Nov 28 AoE which is 14 days.

Reviewer: Nov 14-28 AoE is 15 days; Nov 15-28 AoE would be 14 days

<https://www.timeanddate.com/worldclock/fixedtime.html?msg=Python+Steering+Council+nominations+close&iso=20221129T12>`_
(the end of November 28, 2022 `Anywhere on Earth
<https://www.ieee802.org/16/aoe.html>`_).

The voting period shall be: December 1, 2022 12:00 UTC through
`December 16, 2022 12:00 UTC

> <https://www.timeanddate.com/worldclock/fixedtime.html?msg=Python+Steering+Council+voting+closes&iso=20221216T12>`_

Reviewer 3: December 1, 12:00 UTC to December 16, 12:00 UTC is 15 days.

> ... followed by a two-week
vote.

Should it be 14 days?

Reviewer: I assumed that too based on the text here, but apparently, at least per the vote config below, it actually starts December 1 00:00 UTC, which would be 15.5 days.

Reviewer: It seems kinda strange to use AoE for a start date as opposed to a deadline; I'm not sure I've seen it done. A start date of Dec 01 AoE would be Nov 30 10:00 UTC.

- TBD

Withdrawn nominations:

- TBD

Voter Roll
==========

All active Python core team members are eligible to vote. Active status
is determined as :pep:`described in PEP 13 <13#membership>`
and implemented via the software at `python/voters <https://github.com/python/voters>`_
[#note-voters]_.

Ballots will be distributed based on the the `Python Voter Roll
<https://github.com/python/voters/blob/master/voter-files/>`_ [#note-voters]_
for this election.

While this file is not public as it contains private email addresses, the
`Complete Voter Roll`_ by name will be made available when the roll is
created.

Election Implementation
=======================

The election will be conducted using the `Helios Voting Service
<https://heliosvoting.org>`__.

Configuration
-------------

Short name: ``2023-python-steering-council``

Name: ``2023 Python Steering Council Election``

Description: ``Election for the Python steering council, as specified in PEP 13. This is steering council election for the 2023 term.``

type: ``Election``

Use voter aliases: ``[X]``

Randomize answer order: ``[X]``

Private: ``[X]``

Help Email Address: ``psf-election@python.org``

> Voting starts at: ``December 1, 2022 00:00 UTC``

Reviewer 3: It says 12:00 UTC above, which is correct?

PEP: 8104

> Title: 2023 Term steering council election

Reviewer: ```suggestion
Title: 2023 Term Steering Council election
```

SC is a proper noun, so it certainly doesn't make sense to not capitalize it if "2023 Term" is considered one and capitalized accordingly

Version: $Revision$

> Last-Modified: $Date$

Reviewer: ```suggestion
```

No need for these long-obsolete legacy headers that are no longer part of the PEP template

Reviewer: Already addressed in Adam's simultaneous review

> Version: $Revision$
> Last-Modified: $Date$

Reviewer 4: ```suggestion
```

Author: Ee Durbin <ee@python.org>
Sponsor: Brett Cannon <brett@python.org>
Status: Active
Type: Informational
Content-Type: text/x-rst
Created: 08-Nov-2022

Abstract
========

This document describes the schedule and other details of the December
2022 election for the Python steering council, as specified in
:pep:`13`. This is the steering council election for the 2023 term
(i.e. Python 3.12).

Election Administration
=======================

The steering council appointed the
`Python Software Foundation <https://www.python.org/psf-landing/>`__

> Director of Infrastructure, Ee Durbin to administer the election.

Reviewer: Either

```suggestion
Director of Infrastructure, Ee Durbin, to administer the election.
```

or

```suggestion
Director of Infrastructure Ee Durbin to administer the election.
```

Schedule
========

There will be a two-week nomination period, followed by a two-week
vote.

> The nomination period shall be: November 14, 2022 through November 29, 2022 12:00 UTC

Reviewer: ```suggestion
The nomination period shall be: November 14, 2022 through `November 29, 2022 12:00 UTC
<https://www.timeanddate.com/worldclock/fixedtime.html?msg=Python+Steering+Council+nominations+close&iso=20221129T12>`_
```

Its helpful to link this time, so the viewer can convert it to their own time zone...even with AoW, people still mess this up. Of course, there's other such websites to choose from for this, I merely suggested on.

Author: Settled this by addressing @brettcannon's request to go with AoE and not specify a time for start. We'll start given periods at approximately 12:00 UTC on the stated day.

> (The end of November 28, 2021 `Anywhere on Earth

Reviewer 5: ```suggestion
(The end of November 28, 2022 `Anywhere on Earth
```

Author: Thanks!

> (The end of November 28, 2022 `Anywhere on Earth

Reviewer: ```suggestion
(the end of November 28, 2022 `Anywhere on Earth
```

Spurious captial

<https://www.ieee802.org/16/aoe.html>`_).

> The voting period shall be: December 1, 2022 12:00 UTC through December 16, 2022
> 12:00 UTC (The end of December 15, 2022 `Anywhere on Earth
> <https://www.ieee802.org/16/aoe.html>`_).

Reviewer: ```suggestion
The voting period shall be: December 1, 2022 12:00 UTC through
`December 16, 2022 12:00 UTC
<https://www.timeanddate.com/worldclock/fixedtime.html?msg=Python+Steering+Council+voting+closes&iso=20221216T12>`_
(the end of December 15, 2022
`Anywhere on Earth <https://www.ieee802.org/16/aoe.html>`_).
```

Same here, for maximum clarity and minimum chance of misunderstandings

Candidates
==========

Candidates must be nominated by a core team member. If the candidate
is a core team member, they may nominate themselves.

Nominees (in alphabetical order):

- TBD

Withdrawn nominations:

- TBD

Voter Roll
==========

All active Python core team members are eligible to vote. Active status

> is determined as described in :pep:`PEP 13 <13#membership>`

Reviewer: Since the link is to the _description_ in PEP 13, not PEP 13 itself, I suggest titling it accurately (particularly helpful for a11y), either:

```suggestion
is determined as :pep:`described in PEP 13 <13#membership>`
```

or

```suggestion
is determined as described in :pep:`13#membership`
```

(which will render as "...as described in [PEP 13 § Membership](https://peps.python.org/pep-0013/#membership)."

> and implemented via the software at `python/voters <https://github.com/python/voters>`_ [1]_.

Reviewer: ```suggestion
and implemented via the software at `python/voters <https://github.com/python/voters>`_
[#note-voters]_.
```

Its a good idea to always use auto-numbered footnotes, so they don't break, get out of sync or out of order.

> Ballots will be distributed based on the `The Python Voter Roll for this
> election
> <https://github.com/python/voters/blob/master/voter-files/>`_
> [1]_.

Reviewer: ```suggestion
Ballots will be distributed based on the the `Python Voter Roll
<https://github.com/python/voters/blob/master/voter-files/>`_ [#note-voters]_
for this election.
```

* Fix link text, which implies this points to the voter roll for this specific election, when it in facts points to the voter role directory
* Update the footnote
* Fix very strange breaks

Name: ``2023 Python Steering Council Election``

Description: ``Election for the Python steering council, as specified in PEP 13. This is steering council election for the 2023 term.``

type: ``Election``

Use voter aliases: ``[X]``

Randomize answer order: ``[X]``

Private: ``[X]``

Help Email Address: ``psf-election@python.org``

Voting starts at: ``December 1, 2022 00:00 UTC``

Voting ends at: ``December 16, 2022 12:00 UTC``

This will create an election in which:

* Voting is not open to the public, only those on the `Voter Roll`_ may
  participate. Ballots will be emailed when voting starts.
* Candidates are presented in random order, to help avoid bias.
* Voter identities and ballots are protected against cryptographic advances.

Questions
---------

Question 1
~~~~~~~~~~

Select between ``0`` and ``- (approval)`` answers. Result Type: ``absolute``

Question: ``Select candidates for the Python Steering Council``

Answer #1 - #N: ``Candidates from Candidates_ Section``

Results
=======

- TBD

Copyright
=========

This document has been placed in the public domain.

Complete Voter Roll
===================

Active Python core developers
-----------------------------

> ::

Reviewer: ```suggestion
.. code-block:: text
```

Otherwise, this will be automatically syntax highlighted as Python code

    TBD

> .. [1] This repository is private and accessible only to Python Core

Reviewer: ```suggestion
.. [#note-voters] This repository is private and accessible only to Python Core
```

As above

   Developers, administrators, and Python Software Foundation Staff as it
   contains personal email addresses.

> 
> 
> 
> ..
>   Local Variables:
>   mode: indented-text
>   indent-tabs-mode: nil
>   sentence-end-double-space: t
>   fill-column: 70
>   coding: utf-8
>   End:

Reviewer 4: ```suggestion
```

>   End:

Reviewer: ```suggestion
   contains personal email addresses.
```

Also long obsolete and no longer part of the new PEP template

Reviewer: Also addressed already in Adam's review at the same time
